core/src/main/java/org/apache/gravitino/storage/relational/mapper/OwnerMetaMapper.java

```diff
@@ -0,0 +1,147 @@
+          + OWNER_TABLE_NAME
+          + " ot SET ot.deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
+          + " + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000"
+          + " WHERE EXISTS ("
+          + " SELECT ct.catalog_id FROM "
+          + CatalogMetaMapper.TABLE_NAME
+          + " ct where ct.catalog_id = #{catalogId}  AND ct.deleted_at = 0 AND ot.deleted_at = 0 AND ct.catalog_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT st.catalog_id FROM "
+          + SchemaMetaMapper.TABLE_NAME
+          + " st where st.catalog_id = #{catalogId} AND st.deleted_at = 0 AND ot.deleted_at = 0 AND st.schema_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT tt.catalog_id FROM "
+          + TopicMetaMapper.TABLE_NAME
+          + " tt where tt.catalog_id = #{catalogId} AND tt.deleted_at = 0 AND ot.deleted_at = 0 AND tt.topic_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT tat.catalog_id FROM "
+          + TableMetaMapper.TABLE_NAME
+          + " tat where tat.catalog_id = #{catalogId} AND tat.deleted_at = 0 AND ot.deleted_at = 0 AND tat.table_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT ft.catalog_id FROM "
+          + FilesetMetaMapper.META_TABLE_NAME
+          + " ft where ft.catalog_id = #{catalogId} AND ft.deleted_at = 0 AND ot.deleted_at = 0 AND ft.fileset_id = ot.metadata_object_id"
+          + ")")
+  void softDeleteOwnerRelByCatalogId(@Param("catalogId") Long catalogId);
+
+  @Update(
+      "UPDATE  "
+          + OWNER_TABLE_NAME
+          + " ot SET ot.deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
+          + " + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000"
+          + " WHERE EXISTS ("
+          + " SELECT st.schema_id FROM "
+          + SchemaMetaMapper.TABLE_NAME
+          + " st where st.schema_id = #{schemaId} AND st.deleted_at = 0 AND ot.deleted_at = 0 AND st.schema_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT tt.schema_id FROM "
+          + TopicMetaMapper.TABLE_NAME
+          + " tt where tt.schema_id = #{schemaId} AND tt.deleted_at = 0 AND ot.deleted_at = 0 AND tt.topic_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT tat.schema_id FROM "
+          + TableMetaMapper.TABLE_NAME
+          + " tat where tat.schema_id = #{schemaId} AND tat.deleted_at = 0 AND ot.deleted_at = 0 AND tat.table_id = ot.metadata_object_id"
+          + " UNION "
+          + " SELECT ft.schema_id FROM "
+          + FilesetMetaMapper.META_TABLE_NAME
+          + " ft where ft.schema_id = #{schemaId} AND ft.deleted_at = 0 AND ot.deleted_at = 0 AND ft.fileset_id = ot.metadata_object_id"
+          + ")")
+  void sotDeleteOwnerRelBySchemaId(@Param("schemaId") Long schemaId);
+
+  @Delete(
+      "DELETE FROM "
+          + OWNER_TABLE_NAME
+          + " WHERE deleted_at > 0 AND deleted_at < #{legacyTimeline} LIMIT #{limit}")
+  Integer deleteOwnerMetasByLegacyTimeline(
```

Comment:
This SQL is used in the `RelationalGarbageCollector`, maybe you need add some codes to call this SQL.